extensions/ql-vscode/src/data-extensions-editor/model-details/model-details-data-provider.ts

```diff
@@ -61,12 +61,45 @@ export class ModelDetailsDataProvider
       return [];
     }
   }
+
+  getParent(
+    item: ModelDetailsTreeViewItem,
+  ): ModelDetailsTreeViewItem | undefined {
+    if (isExternalApiUsage(item)) {
+      return undefined;
+    } else {
+      return this.externalApiUsages.find((e) => e.usages.includes(item));
+    }
+  }
+
+  public resolveCanonicalUsage(usage: Usage): Usage | undefined {
+    for (const externalApiUsage of this.externalApiUsages) {
```

Comment:
We could make `resolveCanonicalUsage` and `getParent` more performant by adding the `ExternalApiUsage` object into the "jumpToUsage" message as well. This way we would at least only have to search all usages of a single method, instead of searching all methods.

In terms of actual time saved it's probably milliseconds at most, but it may be noticeable on really big databases.